### ISSUE TITLE
Add ability to trust SSL Certificates

### ISSUE THREAD

**Issue by dpgraham** [appium/appium-desktop#390]
(see https://github.com/electron/electron/pull/9099)

**Comment by gregma** [comment]
Has this been merged yet?  Underling issues seem to have been merged into electron for both win and mac.  Unfortunately, we are blocked by this issue .

**Comment by dpgraham** [comment]
@gregma This hasn't been started yet. What's the use-case that you're blocked on?

**Comment by dpgraham** [comment]
@gregma I'm working on this now. Can I just confirm that the issue is that when you try to start an Appium session, you're getting a Certificate error?

Also, is it via `Custom Server` or `Sauce Labs` session?

**Comment by gregma** [comment]
Here is an actual screenshot of the error:

![2018-03-02](https://user-images.githubusercontent.com/3743375/36985715-dd42f3cc-204c-11e8-9970-90da87cdbd0a.png)

This is with a SauceLabs TestObject session.  

Greg

**Comment by dpgraham** [comment]
@jlipps I'm working on this ticket right now. I'd like to make it so that we can give users the option to ignore certificate errors to handle cases like this one. Do you think we should add a checkbox to the different server types that says something like `Allow untrusted certs` or should it be hidden in some kind of advanced area.

**Comment by dpgraham** [comment]
Hi @gregma,

I think I have it fixed. I tested it locally by setting up an insecure local Appium server and testing connections on it. 

To allow insecure certs, all you need to do is check a box specifying `Allow Unauthorized Certificates`.

Would you have any interest in testing the solution? It would be good to know if it fixes your problem prior to doing a release.

To run it you just need to clone this repo, checkout the branch `https://github.com/appium/appium-desktop/tree/dpgraham-allow-unauthorized` and then run the following commands to run Appium Desktop locally in development mode

```
npm install
npm run dev
```

**Comment by gregma** [comment]
Let me give it a try!

**Comment by gregma** [comment]
Worked just fine!  Ship it!

**Comment by dpgraham** [comment]
@gregma Awesome! I'll ship it with the next release.

**Comment by dpgraham** [comment]
Hey @gregma this is shipped in 1.5.0; sorry for the delay

The UI is a little bit different. Now, the checkbox is hidden in a collapsible menu called `Advanced Settings`.

**Comment by gregma** [comment]
Unfortunately, now there are two boxes.  The one inside Advanced Settings which does work, and the one to the right of "TestObject API Key" (in the "TestObject" tab) and one to the right of "Remote Host" (In the "Custom Server" tab) which doesn't work.

**Comment by gregma** [comment]
![testobject](https://user-images.githubusercontent.com/3743375/38382122-cc2f2cfc-38bd-11e8-8276-07c7dd1dd1d7.png)

**Comment by gregma** [comment]
![customserver](https://user-images.githubusercontent.com/3743375/38382143-de243164-38bd-11e8-83ff-3e820f57152d.png)

**Comment by dpgraham** [comment]
Arg, I forgot to take it out.

Does using `Advanced Settings` still solve your problem, though, @gregma?

**Comment by gregma** [comment]
It does indeed!

**Comment by dpgraham** [comment]
Okay, that's a relief. Yeah I'll take it out for the next release.

Thanks for letting me know!
